go/libraries/doltcore/sqle/enginetest/dolt_queries.go

```diff
@@ -125,6 +125,146 @@ var DescribeTableAsOfScriptTest = queries.ScriptTest{
 	},
 }
 
+var DoltRevisionDbScripts = []queries.ScriptTest{
```

Comment:
Any tests for things like `HEAD~`? What about commit hashes?

Reply:
No support for `HEAD~` refs yet, but I'd be more than happy to add that in a separate PR. Seems like a very natural and useful extension. 

I made BATS tests covering commit hashes in db rev specs, but didn't come up with a good way to get the commit hash and easily use it in the enginetests with the `ScriptTest` structures. I figured the BATS coverage was decent, but I'll try adding a non-ScriptTest and see if I can get some enginetest coverage for commits, too.